### ISSUE TITLE
Implement health indication for crosshair

### ISSUE THREAD

**Issue by JNechaevsky** [fabiangreffrath/crispy-doom#298]
I'm now working on crosshair implementation, and not very happy about it. However, there is a feature that you may find interesting - crosshair can indicate player's health by following logic:

```
if (plyr->health >= 66)
// green color translation
else if (plyr->health >= 33)
// yellow color translation
else
// red color translation
```
NB: health conditions are not very correct, but I hope you'll get the point. In Crispness menu, "Draw Crosshair:" item may have following options:

- Off
- Static
- Static (indicate health)
- Projected
- Projected (indicate health)

My additional recommendation is to don't make it combined with colored HUD elements, it's a different thing. Please also note, that this feature is ZDoomims, but the feature itself is not so bad.

_Suggestion №90_

**Comment by fabiangreffrath** [comment]
Thanks for the suggestion!

> crosshair can indicate player's health by following logic

I'd rather follow the same logic that colorizes the status bar's health widget:
https://github.com/fabiangreffrath/crispy-doom/blob/master/src/doom/st_stuff.c#L1552

> Draw Crosshair:" item may have following options:

I'd rather add an additional item, I guess *sigh* Another possible feature is to colorize the crosshair if it points to an enemy. Also, I am not sure if I want the projected crosshair to represent player's health - it's supposed to be a "laser spot" after all.

> Suggestion №90

Wasn't #197 already counted as No90?

**Comment by JNechaevsky** [comment]
> I'd rather follow the same logic that colorizes the status bar's health widget

Ah, hm, well. Maybe I was wrong, despite of it's a slightly different things _(in my taste*)_, it's a same and proper logic. In any way, if players will not like it and want a separate option, some further actions always can be done.

> I'd rather add an additional item

Yes, and that's what I was trying to do not suggest. :( Sadly, first one Crispness page is already too full to add a brief descriptions (well, there is still a place for exactly one line), as was planned in #236. 

I like the idea of colorizing while aiming at enemy, sounds very, very neat! :+1: As about laser spot - well, some reasonable conservatism is also good.

Ah yes, since I have started to implement my own crosshair, please, don't hurry with this implementation and give few days to play around. I was almost never playing traditional Doom with crosshair, and maybe some ideas will come. But maybe even better to don't involve it into another featureism bridgehead, it's just a crosshair after all.

> Wasn't #197 already counted as No90?

Uh-oh, right, it was a #297. Sorry for inconvenience! Corrected to №91.

**Comment by JNechaevsky** [comment]
> Like, letting player chose at which health values to switch colors?

No, no, no! I was meaning another thing: separate options for colored HUD and colored crosshair. For example, I want to have colored HUD, but not colored (simply red) crosshair. But duh, it's a way to additional menu item >:(

**Comment by Zodomaniac** [comment]
@JNechaevsky I think the best option is to add the crosshair color option to Tactical page. The crosshair presence and shape are defined there.

**Comment by JNechaevsky** [comment]
I'm totally agree that crosshair is a tactical, not simply visual feature.  As about splitting/non splitting... No idea for the moment, sorry. :) Probably, there is no need to split it there are only 2 menu items, but for 3-4 it could be good.

**Comment by JNechaevsky** [comment]
Done with my implementation ([here](https://github.com/JNechaevsky/russian-doom/commit/e9b8edb451d54ecb9a755e0b4137cf5961ab2c7b) and [here](https://github.com/JNechaevsky/russian-doom/commit/1977dddef34a4c83a5230375cbd6d342f57010d5), results are [here](https://github.com/JNechaevsky/russian-doom/issues/64#issuecomment-384768877)). 

Well. From my experience, I'm having three options for crosshair and have moved it into separate gameplay enhancements section. And as I promised, a few raw ideas, which I, frankly, can't recommend for implementation:

1. Something can be done while `iddqd` or invulnerability. Obivious choices will be to colorize crosshair with white/gray color, but no. This way it will be barely notable in inversed colormap. Another idea is to colorize it with gold/orange, but no again - yellow color should be preserved for "middle health" indication.
2. Make a `hires` independent crosshair, i.e. smaller, unscaled. Probably will not be looking good for STCFN chars.
